.github/workflows/tests.yaml

```diff
@@ -30,8 +30,9 @@ jobs:
         uses: charmed-kubernetes/actions-operator@main
         with:
           provider: microk8s
-          channel: 1.25/stable
-          bootstrap-options: '--agent-version=2.9.34'
+          channel: 1.25-strict/stable
+          juju-channel: 3.1
+          bootstrap-options: '--agent-version=3.1.0'
 
       - name: Run integration tests
         # set a predictable model name so it can be consumed by charm-logdump-action
```

Comment:
You should add a job to automatically publish the lib once the charm has been published as well (probably in the `publish.yaml` file). Something like this should work:

```yaml
      - name: Publish libs
        env:
          CHARMCRAFT_AUTH: "${{ secrets.CHARMCRAFT_AUTH }}"
        run: |
          charmcraft publish-lib charms.hydra.v0.hydra_endpoints

```